src/avif.c

```diff
@@ -721,6 +721,54 @@ avifBool avifCleanApertureBoxConvertCropRect(avifCleanApertureBox * clap,
     return AVIF_TRUE;
 }
 
+// ---------------------------------------------------------------------------
+
+avifBool avifAreGridDimensionsValid(avifPixelFormat yuvFormat, uint32_t columns, uint32_t rows, uint32_t tileW, uint32_t tileH, avifDiagnostics * diag)
+{
+    // The rules below only apply to grid images. Single-cell images do not have these constraints.
+    if ((columns == 1) && (rows == 1)) {
+        return AVIF_TRUE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - the tile_width shall be greater than or equal to 64, and should be a multiple of 64
+    //   - the tile_height shall be greater than or equal to 64, and should be a multiple of 64
+    // The "should" part is ignored here.
+    if ((tileW < 64) || (tileH < 64)) {
+        avifDiagnosticsPrintf(diag,
+                              "[Strict] Grid image tile width (%u) or height (%u) cannot be smaller "
+                              "than 64 if there is more than one cell. "
+                              "See MIAF (ISO/IEC 23000-22:2019), Section 7.3.11.4.2",
+                              tileW,
+                              tileH);
+        return AVIF_FALSE;
+    }
+
+    // ISO/IEC 23000-22:2019, Section 7.3.11.4.2:
+    //   - when the images are in the 4:2:2 chroma sampling format the horizontal tile offsets and widths,
+    //     and the output width, shall be even numbers;
+    //   - when the images are in the 4:2:0 chroma sampling format both the horizontal and vertical tile
+    //     offsets and widths, and the output width and height, shall be even numbers.
+    // Only the tile dimensions are enforced since tile offsets and image dimensions depend on them.
+    // If the rules above were not respected, the following problematic situation may happen:
+    //   Some 4:2:0 image is Pi pixels wide and has N cell columns.
+    //   Each cell is Pc pixels wide, Pc being odd.
+    //   The chroma plane of the whole image is Pi/2 or (Pi+1)/2 pixels wide, depending on the rounding.
+    //   The chroma plane of the summed cells is (Pc/2)*N or ((Pc+1)/2)*N pixels wide, depending on the rounding.
+    //   This leads to either N missing pixels or N extra pixels, compared to the whole image.
```

Comment:
If I understand your example correctly, this sentence should read

`This leads to either N/2 or (N-1)/2 extra pixels, compared to the whole image.`

because we are comparing` (Pi+N)/2` with `Pi/2` or `(Pi+1)/2`.